File: Lab04 JAVA/Gerente.java
public class Gerente extends Funcionario implements InterfaceGerente {
    private int numeroDeEquipes;

    public Gerente(String nome, double salarioBase, int numeroDeEquipes) {
        super(nome, salarioBase);
        this.numeroDeEquipes = numeroDeEquipes;
    }

    public int getNumeroDeEquipes() {
        return numeroDeEquipes;
    }

    public void setNumeroDeEquipes(int numeroDeEquipes) {
        this.numeroDeEquipes = numeroDeEquipes;
    }

    public double calcularBonus() {
        return numeroDeEquipes * 500;
    }
    
    @Override
    public String descricao() {
    return super.descricao() + ", Número de equipes: " + numeroDeEquipes +
           ", Bônus: " + calcularBonus();
}

}

File: Labs/Lab04 C++/Gerente.cpp
#include "Gerente.hpp"

using namespace std;

Gerente::Gerente(const string &nome, int salario_base, int numero_de_equipes)
    : Funcionario(nome, salario_base), numero_de_equipes(numero_de_equipes) {}

int Gerente::getNumero_de_equipes()
{
    return numero_de_equipes;
};

int Gerente::bonus() const
{
    return numero_de_equipes * 500;
}